src/java/main/org/apache/zookeeper/server/quorum/QuorumCnxManager.java

```diff
@@ -638,13 +639,22 @@ public void run() {
                     LOG.info("My election bind port: " + addr.toString());
                     setName(addr.toString());
                     ss.bind(addr);
+                    ss.setSoTimeout(10 * 1000); // Ten seconds
```

Comment:
By setting the timeout to 10s the timeout exception will fire potentially every 10s, is that what we want below?